Re-enable TestNamedTensor.test_big_tensor_repr

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29407 Re-enable TestNamedTensor.test_big_tensor_repr**

Fixes https://github.com/pytorch/pytorch/issues/27753.

The bug was that random tensors print subtly differently. This causes
the "names=" tag to appear in slightly different places; sometimes it is
on the same line as the data, sometimes it is on different lines.

For this test, we wanted to know the following:
- printing a big named tensor's repr doesn't crash
- a big named tensor's repr shows the names

This PR changes the test to check those two things.

Test Plan:
- run existing tests

Differential Revision: [D18428657](https://our.internmc.facebook.com/intern/diff/D18428657)

zou3519 committed:
Re-enable TestNamedTensor.test_big_tensor_repr

Fixes https://github.com/pytorch/pytorch/issues/27753.

The bug was that random tensors print subtly differently. This causes
the "names=" tag to appear in slightly different places; sometimes it is
on the same line as the data, sometimes it is on different lines.

For this test, we wanted to know the following:
- printing a big named tensor's repr doesn't crash
- a big named tensor's repr shows the names

This PR changes the test to check those two things.

Test Plan:
- run existing tests

nairbv commented on test/test_namedtensor.py:
```diff
@@ -275,12 +275,11 @@ def test_no_multiprocessing_support(self):
-            expected = "{}, names={})".format(repr(unnamed_tensor)[:-1], named_tensor.names)
-            self.assertEqual(repr(named_tensor), expected)
+            names_tag = 'names={}'.format(named_tensor.names)
+            self.assertTrue(names_tag in repr(named_tensor))
```
nit: we could use assertRegexpMatches so that any error message would print both of the non-matching strings

zou3519 replied:
I'm going to use `self.assertEquals`

zou3519 replied:
It turns out there is something called `assertIn` that solves this problem

zou3519 committed:
Update on "Re-enable TestNamedTensor.test_big_tensor_repr"


Fixes https://github.com/pytorch/pytorch/issues/27753.

The bug was that random tensors print subtly differently. This causes
the "names=" tag to appear in slightly different places; sometimes it is
on the same line as the data, sometimes it is on different lines.

For this test, we wanted to know the following:
- printing a big named tensor's repr doesn't crash
- a big named tensor's repr shows the names

This PR changes the test to check those two things.

Test Plan:
- run existing tests
